### PR TITLE
Set 2.6 as minimum ruby version for Sinatra 3

### PR DESCRIPTION
**Pull request by epergo**
Now that we have an `unstable` branch I think we can start doing these kind of changes. Ruby 2.5 support was dropped last March so I think that for the release of Sinatra 3.0 we could set the required minimum ruby version to 2.6 which will have support until March 2022.

I have cleaned up the `indifferent_hash` implementation because we had a lot of conditionals to check if the `hash` implementation of the installed ruby version had certain methods or not.

WDYT?

### CODE REVIEW

File: lib/sinatra/indifferent_hash.rb [hunk 1]
```diff
@@ -1,11 +1,4 @@
 # frozen_string_literal: true
-$stderr.puts <<EOF if !Hash.method_defined?(:slice) && !$LOAD_PATH.grep(%r{gems/activesupport}).empty? && ENV['SINATRA_ACTIVESUPPORT_WARNING'] != 'false'
```
Comment (epergo) [inline lib/sinatra/indifferent_hash.rb:2]: `Hash#slice` was added in Ruby 2.5 so this `if` is going to be `false` always.

**Comment by jkowens** [comment]
I like this suggestion 👍

**Review by olleolleolle — APPROVED** [review]
A thorough and careful update! Thanks, @epergo!

(Any RuboCop configuration that could get an update, too.)

**Review by namusyaka — APPROVED** [review]

**Comment by olleolleolle** [comment]
@namusyaka Do you feel ready to merge this?

**Pull request merged**